.semaphore/semaphore.yml

```diff
@@ -4,8 +4,14 @@ agent:
   machine:
     type: e1-standard-2
     os_image: ubuntu1804
-blocks:
 
+# enable auto-cancel for branches, tags and pull-requests
+# https://docs.semaphoreci.com/use-cases/auto-cancel-previous-pipelines-on-a-new-push/
+auto_cancel:
+  queued:
+    when: "true"
```

Comment:
how does this end up with destroying resources in a graceful way?

Reply:
this will only cancel queued pipelines that haven't been started yet. https://docs.semaphoreci.com/use-cases/auto-cancel-previous-pipelines-on-a-new-push/